tldw_Server_API/app/api/v1/endpoints/chat.py

```
@@ -4393,6 +4442,46 @@ def _sanitize_json_for_rate_limit(request_json: str) -> str:
+
+    The callback is invoked by chat_service when the stream finishes with the
+    total estimated token count.  It must:
+      - Record billing usage synchronously (apply_usage_delta) since
+        LimitEnforcer.__aexit__ runs at handler return time, before the stream
+        generator fires this callback.
+      - Return the RG governor coroutine so the caller can await it.
+    """
+    if not rg_handle_id and not billing_enforcer:
+        return None
+
+    def _commit(total: int) -> Any:
```

Ruff (0.15.9)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `_commit`

(ANN401)